app/services/reactivate_account_session.rb

```diff
@@ -28,7 +28,7 @@ def suspend
   # @param [Pii::Attributes]
   def store_decrypted_pii(pii)
     reactivate_account_session[:validated_personal_key] = true
-    Pii::Cacher.new(@user, session).save_decrypted_pii(pii)
+    Pii::Cacher.new(@user, session).save_decrypted_pii(pii, @user.password_reset_profile.id)
```

Comment:
I had a little trouble chasing down where this one was called. It is the method that is invoked after a user enters their personal key and their recovery PII is decrypted. The recovery PII is written to the session. The user is sent to the `#new` action of `VerifyPasswordController` and then submits to `#update` which invokes the `#decrypted_pii` method a few lines down to read that PII from the session and re-encrypt it.

The controllers involved here all have a before action that checks to make sure the user does in fact have a password reset profile.

Reply:
nit - Is there a way to make sure that pattern is carried through to future controllers?

Reply:
Is this in regards to the before action? The before action to ensure there is a password reset profile is pretty essential to the recovery of a password reset profile. I would expect anybody operating on this code to ensure some sort of control like that is in place. I am not sure of a good way to enforce that that fits within the scope of this change request. There are lots of other bits of code dependent on the password reset profile spread out across these controllers.